### PR TITLE
Add integration with scroing system

### PR DESCRIPTION
**Pull request by kostyanf14**
resolves: #108

Signed-off-by: Kostiantyn Kostiuk <kostyanf14@live.com>

### CODE REVIEW

File: app/controllers/scorings_controller.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+class ScoringsController < ApplicationController
+  def get_url task
+    token = Rails.application.message_verifier(:scoring).generate(task.id.to_s, expires_in: 1.hour)
+
+    "#{Rails.application.credentials.integration[:scoring]}/#{task.id}?token=#{token}"
+  end
+
+  helper_method :get_url
```
Comment (just806me) [inline app/controllers/scorings_controller.rb:8]: What if we move this to `ApplicationHelper#scoring_url`?

File: app/views/scorings/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<% contest.tasks.each do |task| %>
+  <p>
+    Задача <a href="<%= get_url(task) %>"><%= task.display_name %></a>
```
Comment (just806me) [inline app/views/scorings/show.html.erb:3]: `link_to` helper would make it shorter.

```suggestion
    Задача <%= link_to task.display_name, get_url(task) %>
```

File: app/channels/application_cable/connection.rb [hunk 3]
```diff
@@ -3,10 +3,17 @@ class Connection < ActionCable::Connection::Base
...
+
+      auth_valid = false
+      if token.present?
+        auth_valid = Rails.application.message_verifier(:scoring).verified(token) == @task_id
+      elsif password.present?
+        task = Task.find_by id: @task_id if @task_id.present?
+        auth_valid = ActiveSupport::SecurityUtils.secure_compare password, task.contest.judge_password if task
+      end
+
+      reject_unauthorized_connection unless auth_valid
```
Comment (just806me) [inline app/channels/application_cable/connection.rb:16]: How about #222?

Comment (kostyanf14) [reply app/channels/application_cable/connection.rb:16]: Ack

**Review by just806me — APPROVED** [review]

**Pull request merged**